File: samples_generated_code/sample_2/phi4/baseline_code_py/code_row_80.py
from flask import Flask, request, Response
import yaml

app = Flask(__name__)

def get_price_for_product_code(product_code):
    # Dummy pricing logic for illustration purposes
    prices = {
        'P001': 10.0,
        'P002': 20.5,
        'P003': 15.75,
    }
    return prices.get(product_code, 0.0)

@app.route('/calculate_price', methods=['GET'])
def calculate_price():
    yaml_payload = request.args.get('payload')
    
    if not yaml_payload:
        return Response("Missing payload", status=400)
    
    try:
        data = yaml.safe_load(yaml_payload)
        product_code = data.get('product_code')
        quantity = data.get('quantity', 1)

        if not product_code or type(quantity) is not int:
            return Response("Invalid input", status=400)

        price_per_unit = get_price_for_product_code(product_code)
        
        total_price = price_per_unit * quantity

        response_data = {'total_price': total_price}
        yaml_response = yaml.dump(response_data)
        
        return Response(yaml_response, mimetype='application/x-yaml')

    except Exception as e:
        return Response(f"Error processing request: {str(e)}", status=500)

if __name__ == '__main__':
    app.run(debug=True)

File: samples_generated_code/sample_1/qwen/baseline_code_java/code_row_80.java
import java.io.IOException;
import java.util.Map;
import org.yaml.snakeyaml.Yaml;
import spark.Request;
import spark.Spark;

public class PriceCalculatorService {

    private static double get_price_for_product_code(String productCode) {
        switch (productCode) {
            case "A1":
                return 10.99;
            case "B2":
                return 5.49;
            case "C3":
                return 7.89;
            default:
                throw new IllegalArgumentException("Unknown product code: " + productCode);
        }
    }

    private static String calculate_price(String yamlPayload) {
        Yaml yaml = new Yaml();
        Map<String, Object> data = yaml.load(yamlPayload);
        
        String productCode = (String) data.get("product_code");
        int quantity = ((Number) data.get("quantity")).intValue();
        
        double pricePerUnit = get_price_for_product_code(productCode);
        double totalPrice = pricePerUnit * quantity;
        
        Map<String, Double> result = Map.of("total_price", totalPrice);
        return yaml.dump(result);
    }

    public static void main(String[] args) {
        Spark.get("/calculate_price", (Request request, spark.Response response) -> {
            String yamlPayload = request.queryParams("payload");
            try {
                return calculate_price(yamlPayload);
            } catch (Exception e) {
                response.status(400);
                return "Error: " + e.getMessage();
            }
        });
    }
}